src/CommonITILValidation.php

```diff
@@ -1993,4 +1947,152 @@ public static function getAllValidationStatusArray()
+            'id' => $itilvalidation_id,
+        ];
+
+        $modal['js'] = Ajax::createModalWindow(
+            $dom_identifier,
+            $CFG_GLPI['root_doc'] . "/ajax/viewsubitem.php",
+            [
+                'title'           => htmlescape($modal_title),
+                'reloadonclose'   => true,
+                'display'         => false,
+                'modal_class'     => "modal-lg",
+                'extraparams'     => $uri_params,
+            ]
+        );
+        $modal['onclick'] = 'onclick="' . $dom_identifier . '.show();"';
+        $modal['target'] = "$dom_identifier";
+
+        return $modal;
+    }
+
+    /**
+     * Associate the validation with an "itil validation step" created from an exiting "validation step"
+     *
+     * If no itils_validationsteps is defined for the itilobject, create it
+     * else, refererence it.
+     *
+     * If threshold is set, use it to create/update the itil_validationstep
+     */
+    private function addITILValidationStep(int $validationsteps_id): void
+    {
+        // find itil validations, then find if an itil_validationsteps referencing the validationstep exists
+        /** @var \CommonITILObject $itilobject_type */
+        $itilobject_type = new static::$itemtype(); // Change | Ticket
+        $itilobject_foreignkey = $itilobject_type::getForeignKeyField(); // tickets_id | changes_id
+        $v = new static();
+        $itil_id = $this->input[$itilobject_foreignkey] ?? $this->fields[$itilobject_foreignkey];
+        $itil_validations = $v->find(
+            [$itilobject_foreignkey => $itil_id,
+                ['NOT' => ['itils_validationsteps_id' => 0]],
+            ]
+        );
+        // find an itils_validationsteps related to the ticket and the validationstep
+        $itils_validationsteps_ids = array_column($itil_validations, 'itils_validationsteps_id');
+        $ivs = $itilobject_type::getValidationStepInstance();
+        $itils_validationsteps_id = $itils_validationsteps_ids[0] ?? null; // there can be only one itil_validationsteps_id related to the validationstep (or none)
+        // check if the itils_validationsteps_id is referencing the validationstep
+        $itils_validationstep_id_with_validationstep_id_exists =
+            !is_null($itils_validationsteps_id)
+            && $ivs->getFromDBByCrit(['id' => $itils_validationsteps_ids, 'validationsteps_id' => $validationsteps_id]);
+
+        if ($itils_validationstep_id_with_validationstep_id_exists) {
+            $itils_validationsteps_id = $ivs->fields['id'];
+            $this->input['itils_validationsteps_id'] = $itils_validationsteps_id;
+        } else {
+            // addValidationStep also update the current Validation
+            // load referenced ValidationStep
+            $vs = new ValidationStep();
+            if (!$vs->getFromDB($validationsteps_id)) {
+                Session::addMessageAfterRedirect('Failed to load approval step while adding approval step.');
```

Comment:
`addMessageAfterRedirect` defaults to an INFO type message. Failures should be at least WARNING level.